packages/editor/src/extensions/task-list/task-list.ts

```diff
@@ -193,6 +193,74 @@ export const TaskListNode = TaskList.extend({
             return false;
           }
         }
+      }),
+      new Plugin({
+        /**
+         * What is state
+         * whats is transaction
+         * what is append transaction
+         */
+        key: new PluginKey("task-list-drag-drop"),
+        appendTransaction(transactions, oldState, newState) {
+          const { from: _from, to: _to, $from: _$from } = oldState.selection;
+          const { from, to, $from } = newState.selection;
+
+          const _isTaskItem =
+            newState.doc.slice(_from, _to).content.firstChild?.type.name ===
+            TaskItem.name;
+
+          const isTaskItem =
+            newState.doc.slice(from, to).content.firstChild?.type.name ===
+            TaskItem.name;
+
+          if (_isTaskItem && isTaskItem) {
+            const taskLists = findChildrenByType(newState.doc, thisType, false);
+
+            for (const taskList of taskLists) {
+              const range = {
+                start: taskList.pos,
+                end: taskList.pos + taskList.node.nodeSize
+              };
+
+              const _isWithInList = range.start < _from && range.end > _to;
+              const isWithInList = range.start < from && range.end > to;
+
+              if (_isWithInList && isWithInList && from < _from) {
```

Comment:
This will be too expensive as it iterates on all nodes to find all the task lists and then iterates on all task lists again. Why not get only the affected task list from `newState` based on the selection value? Then you won't need that last `if statement`.